pypodio2/transport.py

```diff
@@ -11,6 +15,45 @@
 import json
 
 
+class _retry:
+    """Exponential backoff decorator for TransportExceptions."""
+
+    def __init__(self, retries, delay, backoff, cap):
+        self.retries = retries
+        self.delay = delay
+        self.backoff = backoff
+        self.cap = cap
+
+    def __call__(self, func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            for attempt in range(self.retries):
+                try:
+                    return func(*args, **kwargs)
+                except TransportException as exc:
+                    status = exc.status.status
+                    is_last_attempt = attempt == self.retries - 1
+                    if is_last_attempt:
+                        raise
+                    elif status >= 500:
+                        time.sleep(self._calc_delay(attempt))
+                        continue
+                    else:
+                        raise
+
+        return wrapper if self._retry_enabled else func
+
+    def _calc_delay(self, attempt):
+        delay = self.delay * self.backoff ** attempt
+        delay = min(delay, self.cap)
+        delay = random.randint(0, delay)
```

Comment:
Any specific reason for only allowing integers? 
```suggestion
        delay = random.random() * delay
```